src/services/storage/models/offer.model.ts

```diff
@@ -112,3 +107,23 @@ export function getBillingPriceAvgQuery (
   )
   `)
 }
+
+/**
+ * This function generates nested query for available capacity for offer
+ */
+export function getAvailableCapacityQuery (): Literal {
+  return literal(`
+    totalCapacity - COALESCE(
+      (SELECT
+        CAST(
+          SUM(
+            "storage_agreement"."size"
+            ) as INTEGER
```

Comment:
Hmm the `INTEGER` part is to remove decimals? I mean I assume that the `size` is a number already anyway no? Otherwise you could not SUM it...

Reply:
True, this can be removed

Reply:
actually, it's better to cast to a string, to make sure that we don't have a problem with big numbers in JS

Reply:
Done, thank you guys